pkg/nfd-worker/nfd-worker.go

```diff
@@ -667,6 +682,14 @@ func (m *nfdWorker) updateNodeFeatureObject(labels Labels) error {
 		}
 		klog.InfoS("creating NodeFeature object", "nodefeature", klog.KObj(nfr))
 
+		// If Spiffe is enabled, we add the signature to the annotations section
```

Comment:
Small note: I think we should add `Namespace: namespace` to the nfr created above so that the "SpiffeData" (and thus the checksum) will be identical in the next discovery cycle when we get the object from the api server (i.e. `!errors.IsNotFound(err)`)